tests/DebugSymbol/TestDebugSymbol.cpp

```diff
@@ -0,0 +1,95 @@
+#include <fstream>
+#include <gtest/gtest.h>
+#include <regex>
+#include <sstream>
+#include <string>
+#include <support/colors.h>
+#include <vector>
+
+#include "llvm/DebugInfo/DWARF/DWARFContext.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "warpo/frontend/Compiler.hpp"
+#include "warpo/passes/DwarfGenerator/DwarfGenerator.hpp"
+#include "warpo/support/FileSystem.hpp"
+#include "warpo/support/Opt.hpp"
+#include "wasm.h"
+
+namespace {
+warpo::cli::Opt<bool> updateFixturesFlag{
+    warpo::cli::Category::All,
+    "-u",
+    "--update-fixtures",
+    [](argparse::Argument &arg) { arg.help("update fixture files instead of comparing").flag(); },
+};
+} // namespace
+
+class TestDebugSymbol_P : public ::testing::TestWithParam<std::string> {};
+
+TEST_P(TestDebugSymbol_P, DebugInfo) {
+  warpo::frontend::init();
+  warpo::frontend::Config config = warpo::frontend::getDefaultConfig();
+  config.useColorfulDiagMessage = false;
+  config.emitDebugInfo = true;
+  Colors::setEnabled(false);
+
+  std::string const testCaseName = GetParam();
+  std::filesystem::path const testDir = std::filesystem::path(__FILE__).parent_path();
+  std::filesystem::path const testFilePath = testDir / (testCaseName + ".ts");
+  std::vector<std::string> const entries{testFilePath.string()};
+  warpo::frontend::CompilationResult const compileResult{warpo::frontend::compile(entries, config)};
+  if (compileResult.errorMessage.size() > 0U) {
+    std::cout << "compile failed due to " << compileResult.errorMessage << std::endl;
+    std::terminate();
+  }
+  std::stringstream ss;
+  ss << *compileResult.m.get();
+  std::string const actual = std::move(ss).str();
+
+  llvm::StringMap<std::unique_ptr<llvm::MemoryBuffer>> debugSections =
+      warpo::passes::DwarfGenerator::generateDebugSections(compileResult.m.variableInfo_.getClassRegistry());
+
+  // Extract debug sections from the compiled module
+  static std::regex const debugSectionPattern{R"(debug_(info|line|str|abbrev|aranges|ranges))"};
+  wasm::Module const *const wasmModule = reinterpret_cast<wasm::Module *>(compileResult.m.get());
+  for (auto const &customSection : wasmModule->customSections) {
+    if (std::regex_match(customSection.name, debugSectionPattern)) {
+      std::unique_ptr<llvm::MemoryBuffer> buffer = llvm::MemoryBuffer::getMemBufferCopy(
+          llvm::StringRef(customSection.data.data(), customSection.data.size()), customSection.name);
+      debugSections[customSection.name] = std::move(buffer);
+    }
+  }
```

Comment:
`debugSections` already have informations from `generateDebugSections`, why we need to copy from custom section